src/api/tasks/tasksApi.ts

```diff
@@ -59,13 +63,59 @@ export const useTasks = (subSite: string, pcolId: number) => {
 export const useAddTasks = (subSite: string, pcolId: number) => {
   const queryClient = useQueryClient();
   const pcol = usePCOL(subSite, pcolId);
+  const documents = useDocuments(subSite, String(pcol.data?.Title));
   const sendEmail = useSendEmail();
   return useMutation({
     mutationFn: async (wfDetails: WorkflowDetails) => {
       const [batched, execute] = subWebContext(subSite).batched();
       const batch = batched.web.lists.getByTitle("tasks");
       let count = 0;
 
+      const document = documents.data?.find(
+        (doc) => doc.ListItemAllFields.DocGroup === "PCOL"
+      );
+      const docRelUrl = document?.ServerRelativeUrl;
+      const attachments = documents.data?.filter(
+        (doc) => doc.ListItemAllFields.DocGroup === "Attachment"
+      );
+
+      let attachmentNames = "";
+      if (attachments) {
+        for (const att of attachments) {
+          attachmentNames += att.Name + "\n";
+        }
+      }
+
+      if (docRelUrl) {
+        PizZipUtils.getBinaryContent(
+          docRelUrl,
```

Comment:
Looks like this library may not handle properly special characters. Example name "PCOL Generation Test #2.docx" tries to get a file named "PCOL%20Genereation%20Test%20". Truncates it at #.